Restructure the documentation

gustawlippa opened the pull request:
This PR updates the documentation so that it is more readable. 

Changes:
- The structure is updated as described in the proposal on Confluence,
- Typos fixed, language improved, issues fixed in multiple files,
- Added admonitions, which are these pretty things: 
![obraz](https://user-images.githubusercontent.com/34194983/132847802-05ea627d-e4a0-4c21-a836-b3d9408e29cb.png)
- Added an installation guide,
- Added content tabs, which allow to choose different versions/options of described content:
![obraz](https://user-images.githubusercontent.com/34194983/133575297-bf095d1c-babc-4d90-8b39-e6e887850bb6.png)
- Renamed the folders to better reflect the content (but I don't know if exactly following the section names is needed, so for now e.g. "Extension modules" section is still inside "modules")
- Added a logo: 
![obraz](https://user-images.githubusercontent.com/34194983/133759231-b1008a95-cff8-4d58-bbe9-f4a68f292431.png)



If you want to see the changes from this PR locally, follow https://squidfunk.github.io/mkdocs-material/getting-started/

vkatsuba commented on README.md:
```diff
@@ -73,22 +74,20 @@ Latest releases:
-    * [Build an iOS messaging app](doc/user-guide/iOS_tutorial.md)
-* [User Guide](https://esl.github.io/MongooseDocs/latest/user-guide/ABCs-of-MongooseIM/). Learn all about how to use MongooseIM in your project. Explore its features, supported XEPs, RFCs and database backends, as well as its architecture and deployment strategies.
-* [Configuration](https://esl.github.io/MongooseDocs/latest/Basic-configuration/). Explore some of the available options including database backend configuration, access control lists, listener and extension modules.
+    * [Build MongooseIM from source code](https://esl.github.io/MongooseDocs/latest/user-guide/How-to-build.md)
```
Not sure that the use link https://esl.github.io/MongooseDocs/latest is correct, cuz it's instead of `latest` will be used added `x.x` tag version and eg for `5.0` and for `5.2` links can be changed or removed etc. Please use relative path as it was before.

NelsonVides reviewed (changes requested):
The main page stopped working, if you just enter the site it says 404.

NelsonVides commented:
> The main page stopped working, if you just enter the site it says 404.

https://nelsonvides.github.io/MongooseIM/latest/ 😕

gustawlippa commented:
Yeah, I've just noticed it as well. It seems that the "Home" page is needed in the end for links to work correctly etc. :/

NelsonVides commented on mkdocs.yml:
```diff
@@ -35,17 +35,18 @@ markdown_extensions:
-      - 'Bootstrap scripts': 'user-guide/Bootstrap-Scripts.md'
-      - 'Client certificate authentication': 'user-guide/client-certificate.md'
   - Tutorials:
       - 'How to Build MongooseIM from source code': 'user-guide/How-to-build.md'
```
I'd reorganise the file locations as well, now we'd have for example the `How-to-build` page in the Tutorials section but in the user-guide directory, which is confusing, also because their filepath is what is later shown in the URL. I think the sections in the site and the directories should match.

gustawlippa replied:
I agree, I was thinking there might have been a reason to have tutorials in the `user-guide` directory, so I left it as it was. But I'm glad you see it this way, will be reorganised.

gustawlippa replied:
I've also renamed "advanced-configuration" to just "configuration"

chrzaszcz commented on doc/getting-started/Quick-setup.md:
```diff
@@ -1,44 +1,22 @@
+    MongooseIM will use its default database - Mnesia, which is faster and simpler to set up, but not intended for production purposes when it comes to persistent data.
 
-Note: It is possible at anytime to use external databases. For more information see the end of this guide.
+!!! Note
+    It is possible at anytime to use external databases. For more information, see the [database backend configuration](../configuration/database-backends-configuration.md) page.
```
Maybe merge with the warning and remove the awkward `at anytime`?

```suggestion
It is possible to use external databases instead - for more information, see the [database backend configuration](../configuration/database-backends-configuration.md) page.
```

chrzaszcz commented on doc/getting-started/Quick-setup.md:
```diff
@@ -154,7 +134,8 @@ mongooseimctl add_rosteritem alice localhost bob localhost bob friends both
 
-Note: The `subs`parameter is the "subscription" to a user's presence. Possible values are: `none`, `from`, `to`, or `both`. A subscription in `both` direction means each user will receive each other's presence.
+!!! Note
+    The `subs`parameter is the "subscription" to a user's presence. Possible values are: `none`, `from`, `to`, or `both`. A subscription in `both` direction means each user will receive each other's presence.
```
```suggestion
    The `subs` parameter is the "subscription" to a user's presence. Possible values are: `none`, `from`, `to`, or `both`. A subscription in `both` direction means each user will receive each other's presence.
```

chrzaszcz commented on doc/migrations/4.2.0_5.0.0.md:
```diff
@@ -3,21 +3,22 @@
 ### Changes in XEPs:
-- `mod_last` 
-   - Table `last` - added server column, updated primary key and indexes.
+
```
Note: the link to `priv/migrations` does note work as it references a file in the repo outside of the docs directory. Maybe replace with a GitHub link?

gustawlippa replied:
I think so. This is the similar case to linking the documentation from README.md. These links will work in GitHub, but a full link should not cause any problems.

chrzaszcz reviewed (commented):
Thanks for this huge bunch of improvements! I only found a few places where it could be improved.

gustawlippa commented on README.md:
```diff
@@ -73,22 +74,20 @@ Latest releases:
+    * [Set up MongooseICE](https://esl.github.io/MongooseDocs/latest/user-guide/ICE_tutorial.md)
+    * [Build an iOS messaging app](https://esl.github.io/MongooseDocs/latest/user-guide/iOS_tutorial.md)
+* [User Guide](https://esl.github.io/MongooseDocs/latest/user-guide/Features/). Learn all about how to use MongooseIM in your project. Explore its features, supported XEPs, RFCs and database backends, as well as its architecture and deployment strategies.
+* [Configuration](https://esl.github.io/MongooseDocs/latest/advanced-configuration/configuration-files/). Explore available options including database backend configuration, access control lists, listener and extension modules.
```
Yes, `advanced` should be dropped - good catch!
I've changed some of the links to be absolute, because:
1. The section is "documentation highlights", so for me it seems it should lead to our hosted documentation.
2. Some of the links were already absolute, so it was really unifying them one way or the other.

chrzaszcz reviewed (approved):
It looks good, I just think it would be nice to let CI run the "build docs" job by removing `skip ci` for the last commit before merging.

mongoose-im commented:
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/74274) / small_tests / c90c6182cdc0efc3f8fbeb9c20498a9491447e5c
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3259/74274/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3259/74274/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2021-09-17_14.30.11/lib.mongooseim.logs/run.2021-09-17_14.30.11/suite.log.html)

NelsonVides reviewed (commented):
I'd have kinda preferred relative links, the CI job would detect if any link doesn't exist. I'm not sure how mkdocs does it for absolute links, if it recognises them as internal or external resources, it might be that it won't throw a warning for non-existing links.

Otherwise, all seems great to me :)

gustawlippa commented:
In the "normal" documentation - of course, there is no need for absolute links. MkDocs does work relatively well with relative links, although it still gets confused and doesn't warn about wrong links sometimes (I fixed a bunch). For example in the `priv/migrations` link that Paweł noticed - no notice from CI, even though it is relative, and this is already live [here](https://esl.github.io/MongooseDocs/latest/migrations/4.2.0_5.0.0/).
The only places where I changed relative to absolute links are in README.md, which is not covered by CI as far as I can tell in any way. I would argue these should point to the documentation that we host, not to a file on GH - as do links literally a few lines above. The other thing is what Paweł pointed out, and it is the other way around - there should be a link from our docs to a specific directory in our repository. If you think so, it can point to a specific commit instead of master, but come on, I don't think this directory will move in the forseeable future, and in this case somebody will have more pressing problems when migrating from versions a few years back than one broken link in the docs.

NelsonVides commented:
> In the "normal" documentation - of course, there is no need for absolute links. MkDocs does work relatively well with relative links, although it still gets confused and doesn't warn about wrong links sometimes (I fixed a bunch). For example in the `priv/migrations` link that Paweł noticed - no notice from CI, even though it is relative, and this is already live [here](https://esl.github.io/MongooseDocs/latest/migrations/4.2.0_5.0.0/).
> The only places where I changed relative to absolute links are in README.md, which is not covered by CI as far as I can tell in any way. I would argue these should point to the documentation that we host, not to a file on GH - as do links literally a few lines above. The other thing is what Paweł pointed out, and it is the other way around - there should be a link from our docs to a specific directory in our repository. If you think so, it can point to a specific commit instead of master, but come on, I don't think this directory will move in the forseeable future, and in this case somebody will have more pressing problems when migrating from versions a few years back than one broken link in the docs.

Ah, only on the README.md, right, got it wrong, I was reading your comments and I thought you meant globally. Good way that way :)

NelsonVides reviewed (approved):
LGTM
